src/main/features/ccj/routes/check-and-send.ts

```diff
@@ -18,28 +22,69 @@ function prepareUrls (externalId: string): object {
 }
 
 function renderView (form: Form<Declaration>, req: express.Request, res: express.Response): void {
+  const user: User = res.locals.user
   res.render(Paths.checkAndSendPage.associatedView, {
     form: form,
-    details: res.locals.user.ccjDraft,
-    amountToBePaid: res.locals.user.claim.totalAmount - (res.locals.user.ccjDraft.paidAmount.amount || 0),
+    details: user.ccjDraft,
+    amountToBePaid: user.claim.totalAmount - (user.ccjDraft.paidAmount.amount || 0),
+    partyAsCompanyOrOrganisation: isPartyCompanyOrOrganisation(user),
     ...prepareUrls(req.params.externalId)
   })
 }
 
+function isPartyCompanyOrOrganisation (user: User): boolean {
+  const claim: Claim = user.claim
+  const type: string = claim.claimData.claimant.type
+  return type === PartyType.COMPANY.value || type === PartyType.ORGANISATION.value
+}
+
+function deserializerFunction (value: any): any {
+  switch (value.type) {
+    case SignatureType.BASIC:
+      return Declaration.fromObject(value)
+    case SignatureType.QUALIFIED:
+      return QualifiedDeclaration.fromObject(value)
+    default:
+      throw new Error(`Unknown declaration type: ${value.type}`)
+  }
+}
+
+function getStatementOfTruthClassFor (user: User): any {
```

Comment:
I dislike the fact this method takes whole objects while decision is made only based on the claimant type. Reducing scope would make the code more readable IMO.

Reply:
Also types could be defined in more strict manner by having:

```
function getStatementOfTruthClassFor (user: User): { new(): Declaration | QualifiedDeclaration } {
```

Reply:
have cleaned it up